ConcurrentModificationException on lookupProvidedValue

### Search first

- [X] I searched and no similar issues were found

### Description

About 7% of application startups fail with a `ConcurrentModificationException` thrown while retrieving the current value of a liquibase configuration. The stack traces are slightly different, but the last part remains the same. Therefore, I think that the error is not caused by the use of the db2i extension but is a core problem. I've stripped the spring part from the stack traces

<details>
<summary>InsertOrUpdateGenerator</summary>

```
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-a::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.command.CommandScope.execute(CommandScope.java:235)
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:237)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Liquibase.runInScope(Liquibase.java:1618)
	at liquibase.Liquibase.update(Liquibase.java:229)
	at liquibase.Liquibase.update(Liquibase.java:213)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:329)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 100 more
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-a::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:123)
	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:102)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:99)
	at liquibase.command.CommandScope.execute(CommandScope.java:212)
	... 112 more
Caused by: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-a::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:761)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:61)
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:111)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:111)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:94)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.Scope.child(Scope.java:264)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:68)
	... 119 more
Caused by: java.util.ConcurrentModificationException
	at java.util.Hashtable$Enumerator.next(Hashtable.java:1531)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.lookupProvidedValue(AbstractMapConfigurationValueProvider.java:100)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.getProvidedValue(AbstractMapConfigurationValueProvider.java:50)
	at liquibase.configuration.LiquibaseConfiguration.getCurrentConfiguredValue(LiquibaseConfiguration.java:140)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:104)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:88)
	at liquibase.configuration.ConfigurationDefinition.getCurrentValue(ConfigurationDefinition.java:62)
	at liquibase.structure.AbstractDatabaseObject.shouldIncludeCatalogInSpecification(AbstractDatabaseObject.java:242)
	at liquibase.structure.core.Schema.hashCode(Schema.java:80)
	at java.util.HashMap.hash(HashMap.java:351)
	at java.util.HashMap.put(HashMap.java:624)
	at java.util.HashSet.add(HashSet.java:231)
	at java.util.AbstractCollection.addAll(AbstractCollection.java:355)
	at liquibase.sql.UnparsedSql.<init>(UnparsedSql.java:40)
	at liquibase.sqlgenerator.core.InsertOrUpdateGenerator.generateSql(InsertOrUpdateGenerator.java:162)
	at liquibase.sqlgenerator.core.InsertOrUpdateGenerator.generateSql(InsertOrUpdateGenerator.java:18)
	at liquibase.sqlgenerator.SqlGeneratorChain.generateSql(SqlGeneratorChain.java:32)
	at liquibase.sqlgenerator.SqlGeneratorFactory.generateSql(SqlGeneratorFactory.java:223)
	at liquibase.util.SqlUtil.getSqlString(SqlUtil.java:361)
	at liquibase.changelog.ChangeSet.lambda$addSqlMdc$2(ChangeSet.java:1555)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:204)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:959)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:524)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:514)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:719)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:245)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:510)
	at liquibase.changelog.ChangeSet.addSqlMdc(ChangeSet.java:1556)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:716)
	... 139 more
```
</details>
<details>
<summary>InsertOrUpdateGenerator2.1</summary>

```
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-b::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.command.CommandScope.execute(CommandScope.java:235)
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:237)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Liquibase.runInScope(Liquibase.java:1618)
	at liquibase.Liquibase.update(Liquibase.java:229)
	at liquibase.Liquibase.update(Liquibase.java:213)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:329)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 100 more
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-b::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:123)
	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:102)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:99)
	at liquibase.command.CommandScope.execute(CommandScope.java:212)
	... 112 more
Caused by: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-b::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:761)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:61)
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:111)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:111)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:94)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.Scope.child(Scope.java:264)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:68)
	... 119 more
Caused by: java.util.ConcurrentModificationException
	at java.util.Hashtable$Enumerator.next(Hashtable.java:1531)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.lookupProvidedValue(AbstractMapConfigurationValueProvider.java:100)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.getProvidedValue(AbstractMapConfigurationValueProvider.java:50)
	at liquibase.configuration.LiquibaseConfiguration.getCurrentConfiguredValue(LiquibaseConfiguration.java:140)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:104)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:88)
	at liquibase.configuration.ConfigurationDefinition.getCurrentValue(ConfigurationDefinition.java:62)
	at liquibase.structure.AbstractDatabaseObject.shouldIncludeCatalogInSpecification(AbstractDatabaseObject.java:242)
	at liquibase.structure.core.Schema.hashCode(Schema.java:80)
	at java.util.HashMap.hash(HashMap.java:351)
	at java.util.HashMap.put(HashMap.java:624)
	at java.util.HashSet.add(HashSet.java:231)
	at java.util.AbstractCollection.addAll(AbstractCollection.java:355)
	at liquibase.sql.UnparsedSql.<init>(UnparsedSql.java:40)
	at liquibase.sqlgenerator.core.InsertOrUpdateGenerator.generateSql(InsertOrUpdateGenerator.java:162)
	at liquibase.sqlgenerator.core.InsertOrUpdateGenerator.generateSql(InsertOrUpdateGenerator.java:18)
	at liquibase.sqlgenerator.SqlGeneratorChain.generateSql(SqlGeneratorChain.java:32)
	at liquibase.sqlgenerator.SqlGeneratorFactory.generateSql(SqlGeneratorFactory.java:223)
	at liquibase.executor.AbstractExecutor.applyVisitors(AbstractExecutor.java:106)
	at liquibase.executor.jvm.JdbcExecutor.access$400(JdbcExecutor.java:40)
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:421)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:160)
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1287)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1269)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:718)
	... 139 more
```
</details>
<details>
<summary>InsertOrUpdateGenerator2.2</summary>

```
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-b::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.command.CommandScope.execute(CommandScope.java:235)
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:237)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Liquibase.runInScope(Liquibase.java:1618)
	at liquibase.Liquibase.update(Liquibase.java:229)
	at liquibase.Liquibase.update(Liquibase.java:213)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:329)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 100 more
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-b::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:123)
	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:102)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:99)
	at liquibase.command.CommandScope.execute(CommandScope.java:212)
	... 112 more
Caused by: liquibase.exception.MigrationFailedException: Migration failed for changeset db/postUpdate.xml::postUpdate-b::author:
     Reason: java.util.ConcurrentModificationException
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:761)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:61)
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:111)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:111)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:94)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.Scope.child(Scope.java:264)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:68)
	... 119 more
Caused by: java.util.ConcurrentModificationException
	at java.util.Hashtable$Enumerator.next(Hashtable.java:1531)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.lookupProvidedValue(AbstractMapConfigurationValueProvider.java:100)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.getProvidedValue(AbstractMapConfigurationValueProvider.java:50)
	at liquibase.configuration.LiquibaseConfiguration.getCurrentConfiguredValue(LiquibaseConfiguration.java:140)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:104)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:88)
	at liquibase.configuration.ConfigurationDefinition.getCurrentValue(ConfigurationDefinition.java:62)
	at liquibase.structure.AbstractDatabaseObject.shouldIncludeCatalogInSpecification(AbstractDatabaseObject.java:242)
	at liquibase.structure.core.Schema.hashCode(Schema.java:80)
	at java.util.HashMap.hash(HashMap.java:351)
	at java.util.HashMap.containsKey(HashMap.java:608)
	at java.util.HashSet.contains(HashSet.java:215)
	at liquibase.sql.UnparsedSql.<init>(UnparsedSql.java:32)
	at liquibase.sqlgenerator.core.InsertOrUpdateGenerator.generateSql(InsertOrUpdateGenerator.java:162)
	at liquibase.sqlgenerator.core.InsertOrUpdateGenerator.generateSql(InsertOrUpdateGenerator.java:18)
	at liquibase.sqlgenerator.SqlGeneratorChain.generateSql(SqlGeneratorChain.java:32)
	at liquibase.sqlgenerator.SqlGeneratorFactory.generateSql(SqlGeneratorFactory.java:223)
	at liquibase.executor.AbstractExecutor.applyVisitors(AbstractExecutor.java:106)
	at liquibase.executor.jvm.JdbcExecutor.access$400(JdbcExecutor.java:40)
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:421)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:160)
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1287)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1269)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:718)
	... 139 more
```
</details>

<details>
<summary>GenerateCheckSum</summary>

```
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.UnexpectedLiquibaseException: java.util.ConcurrentModificationException
	at liquibase.command.CommandScope.execute(CommandScope.java:235)
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:237)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Liquibase.runInScope(Liquibase.java:1618)
	at liquibase.Liquibase.update(Liquibase.java:229)
	at liquibase.Liquibase.update(Liquibase.java:213)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:329)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 100 more
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.UnexpectedLiquibaseException: java.util.ConcurrentModificationException
	at liquibase.changelog.StatusChangeLogIterator.run(StatusChangeLogIterator.java:66)
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:89)
	at liquibase.command.CommandScope.execute(CommandScope.java:212)
	... 112 more
Caused by: liquibase.exception.UnexpectedLiquibaseException: java.util.ConcurrentModificationException
	at liquibase.util.MD5Util.computeMD5(MD5Util.java:33)
	at liquibase.change.CheckSum.compute(CheckSum.java:71)
	at liquibase.change.AbstractChange.generateCheckSum(AbstractChange.java:492)
	at liquibase.changelog.ChangeSet.generateCheckSum(ChangeSet.java:346)
	at liquibase.changelog.ChangeSetStatus.<init>(ChangeSetStatus.java:33)
	at liquibase.changelog.visitor.StatusVisitor.addStatus(StatusVisitor.java:44)
	at liquibase.changelog.visitor.StatusVisitor.skipped(StatusVisitor.java:38)
	at liquibase.changelog.StatusChangeLogIterator$1.run(StatusChangeLogIterator.java:59)
	at liquibase.Scope.lambda$child$0(Scope.java:194)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:193)
	at liquibase.Scope.child(Scope.java:172)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.Scope.child(Scope.java:264)
	at liquibase.changelog.StatusChangeLogIterator.run(StatusChangeLogIterator.java:39)
	... 114 more
Caused by: java.util.ConcurrentModificationException
	at java.util.Hashtable$Enumerator.next(Hashtable.java:1531)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.lookupProvidedValue(AbstractMapConfigurationValueProvider.java:100)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.getProvidedValue(AbstractMapConfigurationValueProvider.java:50)
	at liquibase.configuration.LiquibaseConfiguration.getCurrentConfiguredValue(LiquibaseConfiguration.java:140)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:104)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:88)
	at liquibase.configuration.ConfigurationDefinition.getCurrentValue(ConfigurationDefinition.java:62)
	at liquibase.util.MD5Util.computeMD5(MD5Util.java:31)
	... 128 more
```
</details>

Both mentioned postUpdate changeSets are loadUpdateData statements that update table data from csv files.

I am unable to find out at what point this exception would occur. Maybe you with deeper liquibase understanding have an idea.

### Steps To Reproduce

The changelog gets configured inside a config xml and registered inside the web application context (ommited here)
```xml
<bean id="liquibase" class="liquibase.integration.spring.SpringLiquibase" primary="true">
	<property name="changeLog" value="classpath:db/db-changelog.xml"/>
</bean>
```
The postUpdate changelogs look similar to this
```xml
<changeSet id="postUpdate-b" author="author" runOnChange="true">
	<loadUpdateData tableName="table" separator=";" primaryKey="column,column2,column3"
					relativeToChangelogFile="true" file="migration/file.csv"/>
</changeSet>
```

I can't really reproduce the problem reliably. About 7% of updates fail. A simple reboot of the environment helps and the error will usually no longer occur (for this round of updates).

### Expected/Desired Behavior

No expection is thrown while starting the application with liquibase.

### Liquibase Version

4.21.1
This started after we updated from 3.5.5.2

### Database Vendor & Version

DB2 for i, V7R4M0 and V7R5M0

### Liquibase Integration

Spring application running inside Tomcat

### Liquibase Extensions

liquibase-db2i

### OS and/or Infrastructure Type/Provider

OS/400

### Additional Context

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)
If you can guide me in the right direction 😄

--- a/liquibase-standard/src/main/java/liquibase/configuration/core/ScopeValueProvider.java
+++ b/liquibase-standard/src/main/java/liquibase/configuration/core/ScopeValueProvider.java
@@ -34,8 +34,4 @@ public class ScopeValueProvider extends AbstractConfigurationValueProvider {
         return null;
     }
 
-    protected Properties getSystemProperties() {
-        return System.getProperties();
-    }
-
 }

--- a/liquibase-standard/src/main/java/liquibase/configuration/core/SystemPropertyValueProvider.java
+++ b/liquibase-standard/src/main/java/liquibase/configuration/core/SystemPropertyValueProvider.java
@@ -3,6 +3,7 @@ package liquibase.configuration.core;
 import liquibase.configuration.AbstractMapConfigurationValueProvider;
 
 import java.util.Map;
+import java.util.Properties;
 
 /**
  * Searches for the configuration values in the system properties {@link System#getProperties()}.
@@ -23,6 +24,6 @@ public class SystemPropertyValueProvider extends AbstractMapConfigurationValuePr
 
     @Override
     protected Map<?, ?> getMap() {
-        return System.getProperties();
+        return (Properties) System.getProperties().clone();
     }
 }

